### PR TITLE
support ipv6 address parse

### PR DESCRIPTION
**Pull request by chuailiwu**
support ipv6 address parse

### CODE REVIEW

File: src/test/java/com/alipay/remoting/simpledemo/QuickStartServerAndClient.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 public class QuickStartServerAndClient {
     public static void main(String[] args) throws RemotingException, InterruptedException {
-        RpcServer rpcServer = new RpcServer(9876);
+        RpcServer rpcServer = new RpcServer( 9876);
```
Comment (nobodyiam) [inline src/test/java/com/alipay/remoting/simpledemo/QuickStartServerAndClient.java:25]: Is this a typo?
```suggestion
        RpcServer rpcServer = new RpcServer(9876);
```

File: src/test/java/com/alipay/remoting/demo/BasicUsageDemoByJunitWithIpv6.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+/**
+ * basic usage demo
+ * 
+ * basic usage of rpc client and rpc server with ipv6
+ * 
+ * @author xiaomin.cxm
```
Comment (nobodyiam) [inline src/test/java/com/alipay/remoting/demo/BasicUsageDemoByJunitWithIpv6.java:51]: Is this author info right?

File: src/test/java/com/alipay/remoting/demo/BasicUsageDemoByJunitWithIpv6.java [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+ * @version $Id: BasicUsageDemo.java, v 0.1 Apr 6, 2016 8:58:36 PM xiaomin.cxm Exp $
+ */
+public class BasicUsageDemoByJunitWithIpv6 {
```
Comment (nobodyiam) [inline src/test/java/com/alipay/remoting/demo/BasicUsageDemoByJunitWithIpv6.java:54]: Is this a demo program or a test program? If it's a demo program, I suppose we won't have methods annotated with `@Test`? If it's a test program, then I think the file name should be like `XXXTest.java`?

File: src/test/java/com/alipay/remoting/simpledemo/QuickStartServerAndClient.java (hunk 1 above)
Comment (chuailiwu) [reply src/test/java/com/alipay/remoting/simpledemo/QuickStartServerAndClient.java]: yes

File: src/test/java/com/alipay/remoting/demo/BasicUsageDemoByJunitWithIpv6.java (hunk 3 above)
Comment (chuailiwu) [reply src/test/java/com/alipay/remoting/demo/BasicUsageDemoByJunitWithIpv6.java]: this file is modified from "BasicUsageDemoByJunit", a demo by Junit, so keep the same style.
it has been deleted for it is a unnecessary UT.

File: src/test/java/com/alipay/remoting/rpc/RpcAddressParserTest.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        Url url = parser.parse("127.0.0.1:12200");
+        Assert.assertEquals("127.0.0.1", url.getIp());
+        Assert.assertEquals(12200, url.getPort());
+        url = parser.parse("127.0.0.1:12200?KEY1=VALUE1&KEY2=VALUE2");
+        Assert.assertEquals("127.0.0.1", url.getIp());
+        Assert.assertEquals(12200, url.getPort());
+        Assert.assertEquals(url.getProperties().size(), 2);
+
+        //normal ipv6 url
+        url = parser.parse("0:0:0:0:0:0:0:1:12200");
```
Comment (JervyShi) [inline src/test/java/com/alipay/remoting/rpc/RpcAddressParserTest.java:40]: IPv6 is compatible with many formats, and the test case here needs to be supplemented with more ipv6 addresses in different formats, for example: https://gist.github.com/syzdek/6086792

Comment (chuailiwu) [reply src/test/java/com/alipay/remoting/rpc/RpcAddressParserTest.java:40]: yes，but i think it is too complex, there are many Ipv6 regex available and most of them don't work properly.  like https://rosettacode.org/wiki/Parse_an_IP_Address#Java
current support ip:port format will be ok, the same as the current ability。

**Review by nobodyiam — APPROVED** [review]
lgtm

**Review by JervyShi — APPROVED** [review]
LGTM

**Pull request merged**